test/e2e/virtual/workspaces/virtual_workspace_test.go

```diff
@@ -114,41 +115,25 @@ func TestWorkspacesVirtualWorkspaces(t *testing.T) {
 				defer server.Artifact(t, func() (runtime.Object, error) {
 					return server.kcpClient.TenancyV1alpha1().Workspaces().Get(ctx, testData.workspace2.Name, metav1.GetOptions{})
 				})
-				if err := server.kcpExpect(workspace1, func(w *tenancyv1alpha1.Workspace) error {
```

Comment:
Here the expectation is not on the same apiServer:
- the `Workspace` creation is run against the virtual workspace API server, which at some point delegates to the KCP server to effectively create the corresponding persisted `Workspace` resource.
- the expectation is run against the KCP server. So it tests that a `Workspace` resource with the same name has effectively been persisted in the kcp server.